edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerConfig.cs

```diff
@@ -15,8 +15,10 @@ public class DockerConfig : IEquatable<DockerConfig>
     {
         // This is not the actual docker image regex, but a less strict version.
         const string ImageRegexPattern = @"^(?<repo>([^/]*/)*)(?<image>[^/:]+)(?<tag>:[^/:]+)?$";
-
+        // Check if it's prefix is "$upstream" and replace with environment variable. Should support $upstream for any case.
```

Comment:
as discussed, you want to enforce only $upstream will be supported.  Please update the comment here.